troposphere/static/js/components/common/ui/SelectMenu2.react.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import backbone from 'backbone';
+import _ from 'underscore';
+
+export default React.createClass({
+    displayName: "SelectMenu",
+
+    propTypes: {
+        onSelect: React.PropTypes.func.isRequired,
+        optionName: React.PropTypes.func.isRequired,
+        list: React.PropTypes.oneOfType([
+            React.PropTypes.instanceOf(backbone.Collection),
+            React.PropTypes.array
+        ]),
+        current: React.PropTypes.object
```

Comment:
The hint option is removed, its kind of a hack. It could instead just be a tooltip.